membership-common/src/main/scala/com/gu/memsub/subsv2/reads/ChargeListReads.scala

```diff
@@ -59,7 +59,31 @@ object ChargeListReads {
       delivery: ProductId,
       nationalDelivery: ProductId,
       contributor: ProductId,
-  )
+  ) {
+
+    val productMap: Map[ProductId, Product] = {
+      import Product._
+      Map[ProductId, Product](
+        weeklyZoneA -> Product.WeeklyZoneA,
+        voucher -> Voucher,
+        digitalVoucher -> DigitalVoucher,
+        delivery -> Delivery,
+        nationalDelivery -> NationalDelivery,
+        weeklyZoneA -> WeeklyZoneA,
+        weeklyZoneB -> WeeklyZoneB,
+        weeklyZoneC -> WeeklyZoneC,
+        weeklyDomestic -> WeeklyDomestic,
+        weeklyRestOfWorld -> WeeklyRestOfWorld,
+        digipack -> Digipack,
+        supporterPlus -> SupporterPlus,
+        supporter -> Membership,
+        partner -> Membership,
+        patron -> Membership,
+        contributor -> Contribution,
+      )
+    }
```

Comment:
this map replaces the old "reads" thing, basically we just look up the product for the product id that we've got, which works at runtime, and we don't care about compile time any more as everything could be anything.